### DISCUSSION TITLE
How to make use of texture arrays aka "Onion textures"?

### DISCUSSION THREAD

**Discussion by jdobonb** [gfx-rs/wgpu#6656] [Q&A] [answered]
I'm new in graphics programming I'm confused about the terminology of a texture array (aka "[onion texture](http://chunkstories.xyz/blog/a-note-on-descriptor-indexing/)") vs an array of textures.

I'm making a simple Minecraft clone an I've read that for making greedy meshing and texture tiling easier I should use a texture array (`sampler2Darray` on Vulkan, the one that is limited to only one resolution) but as far as my knowledge goes the example [`texture_arrays`](https://github.com/gfx-rs/wgpu/tree/trunk/examples/src/texture_arrays) is using an array of textures, am I wrong?

Does `wgpu` have support for `sampler2Darray`-like textures? If it does, where can I find more info about how to implement it?

**Comment by cwfitzgerald** [comment, marked as answer]
Yes, take a look at https://docs.rs/wgpu/latest/wgpu/type.TextureDescriptor.html#structfield.dimension and set the dimension to 2D, but add a depth greater than 1. This can then be used with `texture_2d_array<f32>` in the shader

**Reply by jdobonb** [reply]
So I set the `dimension` field to `TextureDimension::2D` and then on the `size` field I set the number of textures in [`depth_or_array_layers`](https://docs.rs/wgpu/latest/wgpu/struct.Extent3d.html#structfield.depth_or_array_layers), am I right?

**Reply by cwfitzgerald** [reply]
Correct.

**Reply by jdobonb** [reply]
Great! Thanks for the help!
